qiskit/transpiler/passes/routing/stochastic_swap.py

```diff
@@ -124,16 +124,15 @@ def run(self, dag):
             self.initial_layout = Layout.generate_trivial_layout(canonical_register)
         # Qubit indices are used to assign an integer to each virtual qubit during the routing: it's
         # a mapping of {virtual: virtual}, for converting between Python and Rust forms.
-        self._qubit_to_int = {bit: idx for idx, bit in enumerate(dag.qubits)}
         self._int_to_qubit = tuple(dag.qubits)
 
         self.qregs = dag.qregs
         logger.debug("StochasticSwap rng seeded with seed=%s", self.seed)
         self.coupling_map.compute_distance_matrix()
-        new_dag = self._mapper(dag, self.coupling_map, trials=self.trials)
+        new_dag = self._mapper(dag, dag, self.coupling_map, trials=self.trials)
         return new_dag
 
-    def _layer_permutation(self, layer_partition, layout, qubit_subset, coupling, trials):
+    def _layer_permutation(self, dag, layer_partition, layout, qubit_subset, coupling, trials):
```

Comment:
Same as with the `commuting_2q_gate_router` this is fine to update because it's an internal private function only used in this pass, so there isn't any backwards compatibility implications with passing a dag here.